src/Microsoft.AspNet.OData/Adapters/WebApiActionMap.cs

```diff
@@ -41,5 +41,29 @@ public bool Contains(string name)
         {
             return this.innerMap.Contains(name);
         }
+
+        /// <summary>
+        /// Gets the action descriptor of the specified action
+        /// </summary>
+        /// <param name="actionName">The name of the action</param>
+        /// <returns>The <see cref="IWebApiActionDescriptor"/> if it exists, otherwise null</returns>
+        public IWebApiActionDescriptor GetActionDescriptor(string actionName)
+        {
+            var group = this.innerMap.FirstOrDefault(g => g.Key == actionName);
+
+            if (group == null)
+            {
+                return null;
+            }
+
+            HttpActionDescriptor descriptor = group.FirstOrDefault();
```

Comment:
@habbes You can get rid of the `if` block above by changing this line to
```suggestion
            HttpActionDescriptor descriptor = group?.FirstOrDefault();
```